File: Scala/src/main/scala/PrimaryClasses/Kex_1sub.scala
package PrimaryClasses

import Golem.GolemEx
import Player.Player

import scala.util.Random

object Kex_1sub {
  private val AttackHit: Int = 110
  private val SkillHit: Int = 100
  private val MagicHit: Int = 70

  def CheckHitOrMiss(hit: Int): Boolean = {
    val value = Random.nextInt(100)
    value <= hit - 1
  }

  def AttackGolem(lv: Int, player: Player): Player = {

    var golem = new GolemEx(
      lv * 50 + 100,
      lv * 10 + 40,
      lv * 10 + 40
    )
    printf("ゴーレムLv.%dが現れた！\n", lv + 1)
    while (golem.Hp > 0) {
      printf("ゴーレム 残りHP：%d\n", golem.Hp)
      print("武器を選択してください（１．攻撃　２．特技　３．魔法）＞")
      val choice = scala.io.StdIn.readInt()
      var damage = 0
      var is_hit = false

      choice match {
        case 1 =>
          damage = 60 + Random.nextInt(41)
          is_hit = CheckHitOrMiss(AttackHit - golem.Flee)
        case 2 =>
          damage = 30 + Random.nextInt(101)
          is_hit = CheckHitOrMiss(SkillHit - golem.Flee)
        case 3 =>
          damage = 20 + Random.nextInt(181)
          is_hit = CheckHitOrMiss(MagicHit - golem.Flee)
        case _ =>
          damage = 0
      }

      if (is_hit) {
        damage -= golem.Defense
        if (damage <= 0) {
          damage = 0
        }
        printf("%dのダメージ！\n", damage)
        golem.Hp -= damage
      } else {
        println("攻撃を外した！")
      }

      println("ゴーレムの攻撃！")
      is_hit = CheckHitOrMiss(golem.Hit)
      if (is_hit) {
        val injury = golem.Attack - player.Defense
        printf("%dのダメージ！\n", injury)
        player.Hp -= injury
        if (player.Hp <= 0) {
          player.Hp = 0
          println("あなたはゴーレムに負けました！ゲームオーバー！")
          return player
        }
      } else {
        println("攻撃を外した！")
      }

      printf("プレイヤー残りHP：%d\n", player.Hp)
      if (golem.Hp <= 0)
        golem.Hp = 0
      if (golem.Hp == 0)
        printf("ゴーレムLv.%dを倒した！\n", lv + 1)
    }
    player
  }
}

File: Java/src/com/ecc/gkgo1a/Kex_1sub.java
package com.ecc.gkgo1a;

import java.util.Random;
import java.util.Scanner;

public class Kex_1sub {
    private static final int ATTACK_HIT = 110;
    private static final int SKILL_HIT = 100;
    private static final int MAGIC_HIT = 70;

    private static boolean CheckHitOrMiss(int hit) {
        Random rand = new Random();
        int val = rand.nextInt(100);
        return val <= hit - 1;
    }

    public static Player AttackGolem(int lv, Player player) {
        GolemEx golem = new GolemEx(
                lv * 50 + 100,
                lv * 10 + 40,
                lv * 10 + 40
        );

        System.out.println(String.format("ゴーレムLv.%dが現れた！", lv + 1));
        try {
            while (golem.Hp > 0) {
                System.out.printf("ゴーレム 残りHP：%d\n", golem.Hp);
                System.out.print("武器を選択してください（１．攻撃　２．特技　３．魔法）＞");
                Scanner reader = new Scanner(System.in);
                int damage = 0;
                boolean is_hit = false;
                Random rand = new Random();
                switch (reader.nextInt()) {
                    case 1:
                        damage = 60 + rand.nextInt(41);
                        is_hit = CheckHitOrMiss(ATTACK_HIT - golem.Flee);
                        break;
                    case 2:
                        damage = 30 + rand.nextInt(101);
                        is_hit = CheckHitOrMiss(SKILL_HIT - golem.Flee);
                        break;
                    case 3:
                        damage = 20 + rand.nextInt(181);
                        is_hit = CheckHitOrMiss(MAGIC_HIT - golem.Flee);
                        break;
                    default:
                        break;
                }

                if (is_hit) {
                    damage -= golem.Defense;
                    if (damage <= 0)
                        damage = 0;
                    System.out.printf("%dのダメージ！\n", damage);
                    golem.Hp -= damage;
                } else {
                    System.out.println("攻撃を外した！");
                }

                System.out.println("ゴーレムの攻撃！");
                is_hit = CheckHitOrMiss(golem.Hit);
                if (is_hit) {
                    int injury = golem.Attack - player.Defense;
                    System.out.printf("%dのダメージ！\n", injury);
                    player.Hp -= injury;
                    if (player.Hp <= 0) {
                        player.Hp = 0;
                        System.out.println("あなたはゴーレムに負けました！ゲームオーバー！");
                        return player;
                    }
                } else {
                    System.out.println("攻撃を外した！");
                }

                System.out.printf("プレイヤー残りHP：%d\n", player.Hp);
                if (golem.Hp <= 0)
                    golem.Hp = 0;
                if (golem.Hp == 0)
                    System.out.printf("ゴーレムLv.%dを倒した！\n", lv + 1);
            }
        } catch (Exception ex) {
            System.err.println(ex.getMessage());
        }
        return player;
    }
}
